conda/core/index.py

```diff
@@ -328,7 +332,65 @@ def maybe_decompress(filename, resp_content):
                              getattr(e.response, 'elapsed', None))
 
 
-def read_local_repodata(cache_path):
+def write_pickled_repodata(cache_path, repodata):
+    # Don't bother to pickle empty channels
+    if not repodata.get('packages'):
+        return
+    try:
+        with open(cache_path + '.q', 'wb') as f:
+            pickle.dump(repodata, f)
+    except Exception as e:
+        import traceback
+        log.debug("Failed to dump pickled repodata.\n%s", traceback.format_exc())
+
+
+def read_pickled_repodata(cache_path, channel_url, schannel, priority):
+    pickle_path = cache_path + '.q'
+    # Don't trust pickled data if there is no accompanying json data
+    if not isfile(pickle_path) or not isfile(cache_path):
+        return None
+    try:
+        with open(cache_path + '.q', 'rb') as f:
+            repodata = pickle.load(f)
+    except Exception as e:
+        import traceback
+        log.debug("Failed to load pickled repodata.\n%s", traceback.format_exc())
+        return None
+    if repodata.get('_url') != channel_url:
+        return None
+    add_pip = context.add_pip_as_python_dependency
+    changed = False
+    # Safe changes to IndexRecord
+    if ((repodata['_add_pip'] != add_pip or
+         repodata['_priority'] != priority or
+         repodata['_schannel'] != schannel)):
+        repodata['_priority'] = priority
+        repodata['_add_pip'] = add_pip
+        changed = True
+        for rec in itervalues(repodata.get('packages', {})):
+            rec.priority = priority
```

Comment:
I don't think it's a real problem. But whatever we do we have to implement it on a per-package basis, not a per-repo basis, because of the situation where packages have been removed from the index but are still present in a user's environment. They need a separate priority value.